Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -228,4 +263,105 @@ theorem mulRingNorm_equiv_padic_of_bounded :
 
 end Non_archimedean
 
+section Archimedean
+
+/-! ## Archimedean case -/
+
+/-! ## Preliminary result -/
+
+/-- Given an two integers `n, m` with `m > 1` the mulRingNorm of `n` is bounded by
+    `m + m * f m + m * (f m) ^ 2 + ... + m * (f m) ^ d` where `d` is the number of digits of the
+    expansion of `n` in base `m`. -/
+lemma MulRingNorm_le_sum_digits (n : ℕ) {m : ℕ} (hm : 1 < m):
+    f n ≤ ((Nat.digits m n).mapIdx fun i _ ↦ m * (f m) ^ i).sum := by
+  set L := Nat.digits m n
+  set L' : List ℚ := List.map Nat.cast (L.mapIdx fun i a ↦ (a * m ^ i)) with hL'
+  -- If `c` is a digit in the expansion of `n` in base `m`, then `f c` is less than `m`.
+  have hcoef {c : ℕ} (hc : c ∈ Nat.digits m n) : f c < m :=
+    lt_of_le_of_lt (MulRingNorm_nat_le_nat c f) (mod_cast Nat.digits_lt_base hm hc)
+  calc
+  f n = f ((Nat.ofDigits m L : ℕ) : ℚ) := by rw [Nat.ofDigits_digits m n]
+    _ = f (L'.sum) := by rw [Nat.ofDigits_eq_sum_mapIdx]; norm_cast
+    _ ≤ (L'.map f).sum := mulRingNorm_sum_le_sum_mulRingNorm L' f
+    _ ≤ (L.mapIdx fun i _ ↦ m * (f m) ^ i).sum := by
+      simp only [hL', List.mapIdx_eq_enum_map, List.map_map]
+      apply List.sum_le_sum
+      rintro ⟨i, a⟩ hia
+      dsimp [Function.uncurry]
+      replace hia := List.mem_enumFrom hia
+      push_cast
+      rw [map_mul, map_pow]
+      apply mul_le_mul_of_nonneg_right (le_of_lt (hcoef _))
+        (pow_nonneg (apply_nonneg f ↑m) i)
+      simp only [zero_le, zero_add, tsub_zero, true_and] at hia
+      refine List.mem_iff_get.mpr ?_
+      use ⟨i, hia.1⟩
+      exact (Eq.symm hia.2)
```

Comment:
We tend to avoid indenting the main part of the proof
```suggestion
    _ ≤ (L.mapIdx fun i _ ↦ m * (f m) ^ i).sum := ?_
  simp only [hL', List.mapIdx_eq_enum_map, List.map_map]
  apply List.sum_le_sum
  rintro ⟨i, a⟩ hia
  dsimp [Function.uncurry]
  replace hia := List.mem_enumFrom hia
  push_cast
  rw [map_mul, map_pow]
  apply mul_le_mul_of_nonneg_right (le_of_lt (hcoef _))
    (pow_nonneg (apply_nonneg f ↑m) i)
  simp only [zero_le, zero_add, tsub_zero, true_and] at hia
  exact List.mem_iff_get.mpr ⟨⟨i, hia.1⟩, hia.2.symm)
```